docs/default.nix

```diff
@@ -54,22 +88,63 @@ pkgs.stdenvNoCC.mkDerivation {
   src = ./.;
 
   patchPhase = ''
+    # The generated documentation has headings at level 2, but we want level 3
+    # so they can be nested under the sections for each module system.
+    REDUCE_HEADINGS='s/^## /### /'
+
+    # The "declared by" links point to a file which only exists when the docs
+    # are built locally. This removes the links.
+    REMOVE_DECLARED_BY='/*Declared by:*/,/^$/d'
+
+    function writeOptions() {
+      if [ -s "$1" ]; then
+        sed -e "$REDUCE_HEADINGS" -e "$REMOVE_DECLARED_BY" <"$1" >>"$2"
+      else
+        echo '*No options provided.*' >>"$2"
+      fi
+    }
+
+    function writeModulePage() {
+      page="options/modules/$1.md"
+      file="src/$page"
+
+      if [ ! -f "$file" ]; then
+        echo "# $1" >>"$file"
+        echo ${lib.escapeShellArg ''
+          > [!NOTE]
```

Comment:
This is not being transformed into a correct note block, despite the call to `substituteInPlace` below.